### PR TITLE
feature: tooltips for timing and statement options

### PR DESCRIPTION
**Pull request by yneves**
Replaced the select by a custom dropdown component that shows help for each option.

### CODE REVIEW

**Comment by yneves** [comment]
image

**Comment by ntgn81** [comment]
Hmm great idea to have help icon. Though having custom elements for each of row might impose performance issues if we display many more rows.

I think we can take your `help icon` idea and put it in header. Don't think we need to have help on each row.

What do you think?

**Comment by yneves** [comment]
@ntgn81 Yeah that's the price to pay, i can optimize it more with some caching. I thought about having the icon on header, but not sure it would fulfil its purpose, which is to assist the user while filling, to select the right choice. Imagine the same many rows, header is far above, out of sight.

**Comment by ntgn81** [comment]
I think that is fine, we will walk them through it and give the right reason. Let's have it on the header for now.

**Comment by yneves** [comment]
And of course, we must measure to be sure the performance concern is real.

File: src/components/dashboard/TodosList.js [hunk 1]
```diff
@@ -100,38 +101,61 @@ class TodosList extends Component {
...
+      label: '§1.97(e)(2)',
+      help: 'All items known < 3 months ago'
+    }];
+    if (matter.timing === '1') {
+      options.push({
+        value: 2,
+        label: 'None',
+        help: 'Neither of the above apply (so pay the fee)'
+      });
+    }
```
Comment (ntgn81) [inline src/components/dashboard/TodosList.js]: Let's move the options outside of class, so we don't have an options array for each instance of this component.

File: src/components/dashboard/DropdownHelp.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return (
+      <div onClick={() => this.setState({ open: !this.state.open })}>
+        <DropdownButton
+          bsSize='xsmall'
+          open={this.state.open}
+          id={this.props.id}
+          title={this.renderTitle()}
+          onSelect={this.props.onChange}
+          onToggle={() => {}}>
+          {this.state.open && this.props.options.map(this.renderOption, this)}
```
Comment (yneves) [inline src/components/dashboard/DropdownHelp.js]: This should minimize rendering time.

**Comment by yneves** [comment]
@ntgn81 pushed changes

**Comment by ntgn81** [comment]
Looks good!

**Pull request merged**